go/vt/vtgate/vindexes/vschema.go

```diff
@@ -97,6 +97,8 @@ type Table struct {
 	Columns                 []Column               `json:"columns,omitempty"`
 	Pinned                  []byte                 `json:"pinned,omitempty"`
 	ColumnListAuthoritative bool                   `json:"column_list_authoritative,omitempty"`
+	ReferencedBy            map[string]*Table      `json:"-"`
+	Source                  *Table                 `json:"-,omitempty"`
```

Comment:
VSchema definition is a `string`, but it's more convenient to have a reference to the actual `*vindexes.Table` here. Handle JSON marshalling in `(*Table).MarshalJSON`.